src/cowboy_swagger.erl

```diff
@@ -90,9 +90,9 @@ add_definition(Name, Properties) ->
   application:set_env(cowboy_swagger, global_spec, NewSpec).
 
 -spec schema(DefinitionName::parameter_definition_name()) ->
-  map().
+  #{binary() => binary()}.
 schema(DefinitionName) ->
-  #{<<"$ref">> => <<"#/definitions/", DefinitionName/binary>>}.
+  #{<<<<"$ref">>/binary>> => <<<<"#/definitions/">>/binary, DefinitionName/binary>>}.
```

Comment:
If the binary is known, it comes up, in dialyzer analysis, something like <<_:8>>. /binary solves this while adding an extra bit of a reading effort. One can always use a predefined type, though, e.g. known_binary()

Reply:
If the binary is known, it comes up, in dialyzer analysis, something like <<_:8>>. /binary solves this while adding an extra bit of a reading effort. One can always use a predefined type, though, e.g. known_binary()

Reply:
Sorry @paulo-ferraz-oliveira, I still don't follow… would you mind showing me the actual warning you're seeing when we don't do the `/binary` thingy?
In general, I'm never in favor of altering our code _just to calm down our tools_. There must be a better way.

Reply:
On OTP 19:
```erlang
src/cowboy_swagger.erl
  92: Type specification cowboy_swagger:schema(DefinitionName::parameter_definition_name()) -> #{binary()=>binary()} is a supertype of the success typing: cowboy_swagger:schema(binary()) -> #{<<_:32>>=><<_:64,_:_*8>>
```

I see two alternatives to _solving_ this issue:
1. it doesn't need solving: we leave the file as-is,
2. we consider that a -spec(). similar to the one proposed is OK: I never did like <<_:32>> instead of binary(), though

I didn't consider the 3 alternative (which is the code I proposed in my PR), since you said you were not in favor of changing code _just to calm down our tools_.

Reply:
Should I revert this specific change, but leave the rest as-is?

Reply:
Yeah, I think I would go with option 1.
I don't think (personally) that underspeccing (I think that's what dialyzer is warning us about when it states that _…is a supertype of…) is a bad thing. Specs are also documentation, and sometimes you want to say "I'll give you a map of binaries" even when you're not returning _any_ map of binaries.

Reply:
The rest of the PR is fine, yes!